### DISCUSSION TITLE
제페토 캐릭터가 발판의 속도를 못 따라가는 문제

### DISCUSSION THREAD

**Discussion by gusriel** [naverz/zepeto-studio-kor#534] [5. Q&A]
애니메이션을 통해 발판을 움직이게 만들었고 캐릭터가 발판의 자식으로 들어가게 만들었는데 캐릭터가 움직이는 발판 속도의 절반 정도만 따라가는 문제가 있습니다. 어떻게 해결하면 좋을까요

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요. 해당 이슈 파악을 위해 
플레이 상태에서 Zepeto Player 인스펙터에서 캐릭터 이동 속도 관련 스크린캡쳐 부탁 드리고,
발판 이동과 관련된 코드 첨부 부탁드립니다.

<img width="368" alt="스크린샷 2022-03-28 오전 10 45 07" src="https://user-images.githubusercontent.com/100118650/160312815-4a3f1191-0b2b-413a-b181-607ec26b9353.png">

**Reply by gusriel** [reply]
![chams](https://user-images.githubusercontent.com/100330442/160316674-17deec61-346d-4491-94e6-d80bf0c27eaf.PNG)
![moving](https://user-images.githubusercontent.com/100330442/160316679-65b5e756-0ab0-44c6-a897-0d22429c409d.PNG)

**Reply by Zepeto-TechEvangelist** [reply]
올려주신 코드에는 parent 세팅만 있고 movingBox의 이동과 관련한 코드는 없습니다.
그 부분 첨부 부탁드립니다.

**Reply by gusriel** [reply]
![movebox1](https://user-images.githubusercontent.com/100330442/160318654-2664c9be-e618-44e4-830d-c9a09cf95d67.PNG)

박스 이동은 애니메이션에 키를 줘서 이동하게 만들었습니다

**Reply by Zepeto-TechEvangelist** [reply]
그렇다면 OnTriggerStay에서 캐릭터 위치, 회전 세팅 아래쪽에
this.myPlayer.additionalRunSpeed = 5;
코드 추가해 보시겠어요?

그리고 OnTriggerExit에서는 
this.myPlayer.additionalRunSpeed = 0;
를 추가해서 원래 속도로 맞춰주세요.

**Reply by gusriel** [reply]
![ezgif-1-4624253189](https://user-images.githubusercontent.com/100330442/160327616-af2d27d5-7f80-4194-93ad-be95924e4bf6.gif)

부모 오브젝트가 움직이는 만큼 자녀가 된 캐릭터가 움직이질 않아요

**Reply by Zepeto-TechEvangelist** [reply]
캐릭터가 올라가서 이동하는 발판의 형태가 아니네요. 
용도가 어떻게 되나요?

**Reply by x4cf** [reply]
자식으로 설정된 제페토 캐릭터의 행렬이 현재 움직이고있는 부모 오브젝트의 각도나 스케일의 영향을 받는것으로 보입니다.
제페토 캐릭터의 부모 오브젝트를 000각도와 111스케일을 가진 오브젝트로 해주셔야할것같습니다.

1.애니메이션의 오브젝트A 포지션을 받아서 부모가 없이 단독으로 동일하게 움직이는 오브젝트B를 제작
2.제페토 캐릭터를 오브젝트B의 자식으로 설정

원하는 기능처럼 안될지도 모르지만 한번 해보는것도 좋을것같아요 :D

**Reply by gusriel** [reply]
테스트용으로 만든 발판입니다 저 위에 캐릭터를 올려도 사진처럼 속도 차이 때문에 떨어져요.

발판 위에 부모 오브젝트를 하나 더 만들어서 A밑에 B밑에 캐릭터를 자식으로 넣는 방법도 해보았으나 똑같았습니다.

**Reply by x4cf** [reply]
확인해보니 유니티 캐릭터 컨트롤러가 위치를 조금씩 수정하면서 이동거리에 차이가 생기는군요

https://cheesafizz.notion.site/f4b997d8384347ff9922fc770bc2d864
클라이언트 스타터
마스터타이머
무빙큐브
를 위의 사이트에서 다운로드 받아서 코드를 참고해보는것도 좋을것같습니다 :D

제작과정 1
https://www.youtube.com/watch?v=U9PNr2VNc34
제작과정 2
https://www.youtube.com/watch?v=WnZ3lz3JuiA

**Comment by gusriel** [comment]
유니티상에서 위치를 못 따라잡아도 제페토상에선 문제 없이 발판을 밟고 속도를 따라가네요 해결됐습니다
